app/models/recipe.rb

```diff
@@ -12,4 +12,17 @@ def notify
     ActionCable.server.broadcast 'notifications',
                                   message: "<p>#{self.title} was created!</p>"
   end
+
+  def fork(user)
+    forked_recipe = self.dup
+    forked_recipe.user = user
+    forked_recipe.original_recipe_id << self.id
+    forked_recipe.title = 'Forked ' + forked_recipe.title
+    forked_recipe.forked_recipes_ids = []
+    forked_recipe.save
```

Comment:
I would create an `attributes` object and use it to update the new fork. (after you `.dup`'ed it
```ruby
attributes = {user: user, original_recipe_id: self.id, title: 'Forked ' + forked_recipe.title, forked_recipes_ids: []}
forked_recipe.uppdate_attributes(attributes)
```

Reply:
:+1: